app/src/main/java/com/hedvig/app/feature/genericauth/otpinput/OtpInputActivity.kt

```diff
@@ -0,0 +1,117 @@
+import androidx.compose.material.Scaffold
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import com.google.accompanist.insets.systemBarsPadding
+import com.hedvig.app.BaseActivity
+import com.hedvig.app.R
+import com.hedvig.app.feature.loggedin.ui.LoggedInActivity
+import com.hedvig.app.ui.compose.composables.ErrorDialog
+import com.hedvig.app.ui.compose.composables.appbar.TopAppBarWithBack
+import com.hedvig.app.ui.compose.theme.HedvigTheme
+import com.hedvig.app.util.extensions.compatSetDecorFitsSystemWindows
+import com.hedvig.app.util.extensions.openEmail
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import org.koin.core.parameter.parametersOf
+
+class OtpInputActivity : BaseActivity() {
+
+    val model: OtpInputViewModel by viewModel {
+        parametersOf(
+            intent.getStringExtra(OTP_ID_EXTRA) ?: throw IllegalArgumentException(
+                "Programmer error: Missing OTP_ID in ${this.javaClass.name}"
+            ),
+            intent.getStringExtra(CREDENTIAL_EXTRA) ?: throw IllegalArgumentException(
+                "Programmer error: Missing CREDENTIAL in ${this.javaClass.name}"
+            )
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        window.compatSetDecorFitsSystemWindows(false)
+
+        setContent {
+            val scaffoldState = rememberScaffoldState()
+
+            HedvigTheme {
+                Scaffold(
+                    topBar = {
+                        TopAppBarWithBack(
+                            onClick = ::onBackPressed,
+                            title = stringResource(R.string.login_navigation_bar_center_element_title)
+                        )
+                    },
+                    scaffoldState = scaffoldState,
+                    modifier = Modifier.systemBarsPadding(top = true),
+                ) {
+                    val viewState by model.viewState.collectAsState()
+                    val events = model.eventsFlow.collectAsState(initial = null)
+                    val openDialog = remember { mutableStateOf(false) }
+
+                    LaunchedEffect(events.value) {
+                        when (events.value) {
+                            is OtpInputViewModel.Event.Success -> startLoggedIn()
+                            OtpInputViewModel.Event.CodeResent -> launch {
+                                delay(1000)
+                                val message = getString(R.string.login_snackbar_code_resent)
+                                scaffoldState.snackbarHostState.showSnackbar(message)
```

Comment:
How does this snackbar leave? When a new event state comes in cancelling the previous coroutine job? Do we want it to stay there forever? Otherwise can we do something like
```kotlin
withTimeout(5000) {
    scaffoldState.snackbarHostState.showSnackbar(message)
}
```
To show it for 5000 (or whatever) milliseconds?

Reply:
It disappears automatically